### PR TITLE
Logstash logging

### PR DESCRIPTION
**Pull request by cableman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,10 +94,20 @@ open http://$(docker-compose port nginx 80)/admin
 ```
 
 Next access the React frontend where `x` below is the id of the configuration entity you just created in the administrative interface.
-```
+```sh
 open http://$(docker-compose port nginx 80)?id=x
 ```
 
+## Loggin
```
Comment (tuj) [inline README.md:101]: Logging

File: README.md [hunk 2]
```diff
@@ -94,10 +94,20 @@ open http://$(docker-compose port nginx 80)/admin
...
+The engine use logstash to log messages and these can be seen in the docket setup with the following command.
```
Comment (tuj) [inline README.md:102]: The engine **uses** logstash to log messages and these can be seen in the **docker** setup with the following command

File: engine/plugins/logger/logger.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
+/**
+ * @file
+ * This is a wrapper class to handel the system logger.
```
Comment (tuj) [inline engine/plugins/logger/logger.js:3]: "handel" => "handle"

File: engine/plugins/logger/logger.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+ * Send message to log-stash.
+ *
+ * @param {string} level
+ *   Log level for the message.
+ * @param {string} message
+ *   The message to send to the logger.
+ */
+Logger.prototype.send = function send(level, message) {
+    if (logstashClient !== null) {
+        // This is for to support legacy log messages.
```
Comment (tuj) [inline engine/plugins/logger/logger.js:43]: "This is for to support legacy log messages." => "This is added to support legacy log messages."

File: engine/plugins/logger/logger.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+ * @param {array} options
+ *   Options defined in app.js.
+ * @param {array} imports
+ *   The other plugins available.
+ * @param {function} register
+ *   Callback function used to register this plugin.
+ */
+module.exports = function(options, imports, register) {
+    // Add event listeners to logging events on the bus. For some reason they need
+    // to have a inner function to work!
```
Comment (tuj) [inline engine/plugins/logger/logger.js:101]: a => an

**Review by tuj — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (cableman) [reply README.md:101]: Fixed

File: README.md (hunk 2 above)
Comment (cableman) [reply README.md:102]: Fixed

File: engine/plugins/logger/logger.js (hunk 3 above)
Comment (cableman) [reply engine/plugins/logger/logger.js:3]: Fixed

File: engine/plugins/logger/logger.js (hunk 4 above)
Comment (cableman) [reply engine/plugins/logger/logger.js:43]: Fixed

File: engine/plugins/logger/logger.js (hunk 5 above)
Comment (cableman) [reply engine/plugins/logger/logger.js:101]: Fixed

**Review by tuj — APPROVED** [review]

**Pull request merged**